libs/pages/logs/application/src/lib/page-application-logs.tsx

```diff
@@ -0,0 +1,98 @@
+import { Environment, Log } from 'qovery-typescript-axios'
+import { useCallback, useMemo, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { useParams } from 'react-router-dom'
+import useWebSocket from 'react-use-websocket'
+import { selectApplicationById } from '@qovery/domains/application'
+import { selectEnvironmentById } from '@qovery/domains/environment'
+import { useAuth } from '@qovery/shared/auth'
+import { ApplicationEntity, LoadingStatus } from '@qovery/shared/interfaces'
+import { LayoutLogs, Table } from '@qovery/shared/ui'
+import { useDocumentTitle } from '@qovery/shared/utils'
+import { RootState } from '@qovery/store'
+import Row from './ui/row/row'
+
+export function PageApplicationLogs() {
+  const { organizationId = '', projectId = '', environmentId = '', applicationId = '' } = useParams()
+
+  const environment = useSelector<RootState, Environment | undefined>((state) =>
+    selectEnvironmentById(state, environmentId)
+  )
+
+  const application = useSelector<RootState, ApplicationEntity | undefined>((state) =>
+    selectApplicationById(state, applicationId)
+  )
+
+  useDocumentTitle(`Application logs ${application ? `- ${application?.name}` : '- Loading...'}`)
+
+  const [logs, setLogs] = useState<Log[]>([])
+  const [loading, setLoading] = useState<LoadingStatus>('not loaded')
+
+  const [pauseLogs, setPauseLogs] = useState(false)
+
+  const { getAccessTokenSilently } = useAuth()
+
+  const applicationLogsUrl: () => Promise<string> = useCallback(async () => {
+    const url = `wss://ws.qovery.com/service/logs?organization=${organizationId}&cluster=${environment?.cluster_id}&project=${projectId}&environment=${environmentId}&service=${applicationId}`
+    const token = await getAccessTokenSilently()
+
+    return new Promise((resolve) => {
+      environment?.cluster_id && resolve(url + `&bearer_token=${token}`)
+    })
+  }, [organizationId, environment?.cluster_id, projectId, environmentId, applicationId, getAccessTokenSilently])
+
+  useWebSocket(applicationLogsUrl, {
+    onOpen: () => setLoading('loaded'),
+    onMessage: (message) => !pauseLogs && setLogs((prev: Log[]) => [...prev, JSON.parse(message?.data)]),
```

Comment:
Here, I have the sensation that this could lead to skipping some message:
If you receive some message when the logs are in pause, when you unpause, only the new socket message will be added to existing logs but the one that you received while you were in pause are lost. No ?

Reply:
Nice catch! I fixed that with a local set it should be better !!